doc/manual/src/glossary.md

```diff
@@ -46,9 +46,31 @@
 
   [realise]: #gloss-realise
 
-- [content-addressed derivation]{#gloss-content-addressed-derivation}
+- [content-address]{#gloss-content-address}
 
-  A derivation which has the
+  Information which is *content-addressed* is referenced in a way that is solely determined by its content.
+  Recreating that information scratch in any way will still result in the same reference.
+  (Deduplicating identical data using content-addressed is usually possible and done, also.)
+
+  Wikipedia has [article](https://en.wikipedia.org/wiki/Content-addressable_storage) about this.
+
+- [input-addressed store object]{#gloss-input-addressed-store-object}
+
+  A store object produced by building a
+  non-[content-addressed](#gloss-content-addressed-derivation),
+  non-[fixed-output](#gloss-fixed-output-derivation)
+  derivation.
+
+- [output-addressed store object]{#gloss-output-addressed-store-object}
+
+  A [store object] whose [store path] is determined by its contents.
+  This includes derivations, the outputs of [content-addressed derivations](#gloss-content-addressed-derivation), and the outputs of [fixed-output derivations](#gloss-fixed-output-derivation).
+
+- [(floating) content-addressing derivation]{#gloss-content-addressed-derivation}
+  A derivation whose outputs are will content addressed.
+  The "floating" adjective it often used to distinguish ones where the output content-address is not known in advanced (like reigular input-0afrom fixed output derivations,
+  Technically, this
+  which has the
```

Comment:
This can be a separate PR. I find the current change quite messy and hard to follow, so cleaning that up is worth an entire review round on its own.

Reply:
oh yeah oops I didn't mean to commit this. The current situation is totally broken, it will need more work.